### PR TITLE
[FEATURE] Add serialization logic to Expectation models

### PR DESCRIPTION
**Pull request by NathanFarmer**
1. Add serialization logic to Expectation models
2. Apply spacing to model titles

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

### CODE REVIEW

File: tests/expectations/core/test_core_model_serialization.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        if isinstance(expectation, MetaExpectation)
+    ]
+    # are they still there?
+    assert len(all_models) > 50
+    for model in all_models:
+        model.schema_json()
+
+
+@pytest.mark.unit
+def test_schema_title():
```
Comment (dctalbot) [inline tests/expectations/core/test_core_model_serialization.py]: I would probably paramtrize this over all core expectations. What happens with a name like ExpectJSONToBeFoo ?

**Review by dctalbot — APPROVED** [review]

File: tests/expectations/core/test_core_model_serialization.py (hunk 1 above)
Comment (NathanFarmer) [reply tests/expectations/core/test_core_model_serialization.py]: I will create mock Expectations with edge cases so we don't have to test everything.

Comment (NathanFarmer) [reply tests/expectations/core/test_core_model_serialization.py]: ✅
